src/elements/content-sidebar/withSidebarAnnotations.js

```diff
@@ -0,0 +1,190 @@
+
+            // TODO: need to address in follow on -- currentUser may be undefined here but is never fetched for sure until ActivitySidebar
+            if (!currentUser) {
+                throw getBadUserError();
+            }
+
+            const feedAPI = api.getFeedAPI(false);
+            const { items: hasItems } = feedAPI.getCachedItems(fileId) || {};
+            const { current } = this.sidebarPanels;
+
+            // If there are existing items in the cache for this file, then patch the cache with the new annotation
+            // If there are no cache entry for feeditems, then it is assumed that it has not yet been fetched.
+            if (hasItems) {
+                feedAPI.addAnnotation(file, currentUser, annotation, requestId, isPending);
+            }
+
+            if (isAnnotationsPath && isOpen && current && typeof current.refresh === 'function') {
+                // If the sidebar is currently open, then force the sidebar to refresh with the updated data
+                current.refresh(false);
+            }
+        }
+
+        updateActiveAnnotation = () => {
+            const {
+                annotatorState: { activeAnnotationId },
+                file,
+                getAnnotationsMatchPath,
+                getAnnotationsPath,
+                history,
+                location,
+            } = this.props;
+            const match = getAnnotationsMatchPath(location);
+            const currentFileVersionId = getProp(file, 'file_version.id');
+            const fileVersionId = getProp(match, 'params.fileVersionId', currentFileVersionId);
+            const newLocationState = activeAnnotationId
+                ? { ...location.state, open: true, silent: false }
+                : location.state;
+
+            // Update the location pathname and open state if transitioning to an active annotation id, force the sidebar open
+            history.push({
+                ...location,
+                pathname: getAnnotationsPath(fileVersionId, activeAnnotationId),
+                state: newLocationState,
+            });
+        };
+
+        updateActiveVersion = () => {
+            const {
+                api,
+                file,
+                fileId,
+                getAnnotationsMatchPath,
+                getAnnotationsPath,
+                history,
+                location,
+                onVersionChange,
+            } = this.props;
+            const feedAPI = api.getFeedAPI(false);
+            const match = getAnnotationsMatchPath(location);
+            const currentFileVersionId = getProp(file, 'file_version.id');
+            const fileVersionId = getProp(match, 'params.fileVersionId');
+            const { items: feedItems = [] } = feedAPI.getCachedItems(fileId) || {};
+            const version = feedItems
+                .filter(item => item.type === 'file_version')
+                .find(item => item.id === fileVersionId);
+
+            if (version) {
```

Comment:
Should we do this check before we derive all the other stuff above?

Reply:
Not sure how much we save -- maybe just `const currentFileVersionId = getProp(file, 'file_version.id');`? In order to get the version, we still need to get `match`, and then use that to determine `fileVersionId`, etc...